Rest

exi opened the pull request:
introduce the rest grid and update all the necessary parts

Commit:
Added Default_IndexController->_storeCurrentProjectId($projectId)

This is a better alternative to public setCurrentProjectId, which looks at the
request itself.
Also deprecated setCurrentProjectId.

Commit:
Added a metadatastore

Commit:
Copied Grid to LegacyGrid to allow further development

Commit:
added abstract rest controller and rest routing

Commit:
Added Todo_TodoController extends Phprojekt_Restcontroller

Commit:
Added metadataAction to IndexController

Commit:
make rest-grid work and use it in Todo

Commit:
recompiled js

Commit:
just save the filters instead of making a request

Commit:
remove unused code

Commit:
make in-line editing work (no rights checking on the client yet)

Commit:
stop polluting global scope

Commit:
implemented putAction

Commit:
Workaround to allow inline editing.

Because of a bug that is explained on
http://dojo-toolkit.33424.n3.nabble.com/DataGrid-JsonRest-Unwanted-GET-request-after-editing-cell-td3477425.html
the grid is not updated after saving. We just refresh the query to force this
for now. This is no good solution, but I can't find a better workaround.

Commit:
don't call the updateData function if it does not exist

Commit:
Merge remote-tracking branch 'origin/count' into rest

* origin/count:
  use correct count function in calendar2_models_calendar2
  added Phprojekt_Tree_Node_Database->getRecordsCount($model, $where)
  fixed some comments
  Added Phprojekt_Item_Abstract->count($where)
  Rewrote Phprojekt_ActiveRecord_Abstract->count().

Commit:
Added Phprojekt_Model_Converter and used it in Phprojekt_RestController

This converter simply converts objects that implement Phprojekt_Model_Interface
into arrays. This makes it quite different from
Phprojekt_Converter_{Csv,Json,...} that also do a lot of other things.

The aim of this new class is to separate different functionalities and make each
part simpler.

Commit:
use Phprojekt_Module::saveTypeIsGlobal instead of $model->hasField(projectId)

Commit:
call Phprojekt:::setCurrentProjectId in RestController->preDispatch if projectId is given

Commit:
Made js<->pseudo-iso conversions a little more robust

Commit:
use localized time display in the grid

Commit:
use the new grid in the calendar

Commit:
remove the updateCacheData function

Commit:
correctly handle datetime values, stop using phprojekt subclasses and hand in functions instead

Commit:
Use Phprojekt_CompressedSender::send instead of echo in Phprojekt_RestController

Commit:
Added comments to Phprojekt_RestRoute

Commit:
Use Phprojekt_CompressedSender instead of echo in IndexController

Commit:
Use Zend_Controller_Action_Exception instead of the removed Phprojekt_PublishedException

Commit:
throw and exception instead of silently using defaults if projectId is not given in phpr.MetadataStore::metadataFor

Commit:
Call phpr.handleError in phpr.MetadataStore::metadataFor if the request fails

Commit:
use this.main.module instead of phpr.module

Commit:
Added doccomments

Commit:
Alignment, style, etc

Commit:
Merge remote-tracking branch 'mayflower/next' into rest

* mayflower/next:
  Fixed review remarks
  fix wrong exception in test
  fix minor failing tests
  remove the published exception
  change all published exceptions into matching Controller_Action_Exceptions
  add templates for error messages
  fix timecard delete button display
  checkstyle
  remove the old timecard update code.
  let the timecard hook into the project cache update.
  remove update form in timecard form.js
  make getProjectRange deferred,use the timecard store to manage favorites in form
  rename timecard booking store and use the merged favorites in timecard main.js
  add favorites functionality to the timecard store
  let timecard use deferreds internally
  let timecard track whether it already has data

Conflicts:
	phprojekt/application/Calendar2/Models/Calendar2.php
        Conflict in comments only

Commit:
use delete instead of "= undefined"

Commit:
Minor fixes from Review

Commit:
Fix a broken call.

metadataFor was modified to break if no projectId is provided, but the old code
still called it without one.

Commit:
Added dummy Calendar2_Models_Calendar2->setRecurrence

This was neccessary because activerecord thinks the virtual "recurrence"-field
does not exist if the setter isn't defined.

Commit:
Set the projectId on getAction if an existing object is requested.

In this case, the conversion needs the projectId.

Commit:
Fall back to projectId 1 if we don't have a current projectId

Commit:
Merge branch 'next' into rest

Conflicts:
	phprojekt/application/Calendar2/Views/dojo/scripts/Main.js
	phprojekt/htdocs/dojo/util/buildscripts/profiles/phprojekt.profile.js

Commit:
fix use of getRecordsFor function

the parameter list was wrong.

Commit:
remove phprojekt_publishedexception

these where replace with the zend_controller_action_exception

Commit:
change restcontroller json encoder to the right encoder

we just want to use Zend_Json

Commit:
fix restcontroller bug with global modules

global modules did not work when a project id was provided.

we now handle this case gracefully and don't throw an exception anymore

Commit:
change project module to use the new grid

add the matching controller and use the rest grid

Commit:
change note to use the rest grid

add controller and use rest grid

Commit:
use rest grid for minutes

add rest controller and change frontend grid.

Commit:
use rest grid in helpdesk

add rest controller and change the frontend grid name

Commit:
use rest grid in filemanager

add rest controller and use rest grid in frontend

Commit:
add non submitting checkbox

this will later be used to prevent the rest grid from updating values
if the checkbox is checked

Commit:
fix grid checkboxes

the checkboxes did not work because the submitted an ajax request upon
clicken which failed

Commit:
fix grid updating

now only update the changed row and not the whole grid

Commit:
make module designer grid a legacy grid

otherwise the new modules will not work

Commit:
Merge branch 'next' into rest

removed version from Calendar Main.js

Conflicts:
	phprojekt/application/Calendar2/Views/dojo/scripts/Main.js

voidus commented:
bamp

Commit:
review fixes

review fixes

Commit:
fix filters for the rest controller

The filters did not work because the filters function called the wrong getModelObject function.
Also the item count was wrong because we did not passed the filters to the tree class.

voidus commented:
Fails tests and doesn't apply. Tests might be fixed by merging voidus/fixTravis.

Commit:
remove webpath from grid.js

this will prevent merge confricts

Commit:
Merge branch 'next' into rest

I just used the Grid.js changes from the rest branch.
All conflicts where triggered by the removed webpath.

Conflicts:
	phprojekt/application/Default/Views/dojo/scripts/Grid.js

Commit:
remove additional occurrences of phpr.webpath

these do not work anymore.

Commit:
fix bug when calling getisotime

we just passed "8" as hour into getisotime, which oly accepts 08:02 format.
This converts it to the double digit form.

exi commented:
merged next and fixed occurring bugs.
Should be ok now. Lets hope travis agrees.
